File: src/main/java/me/waiaf/skytizencore/listeners/ComposterInteractEvent.java
package me.waiaf.skytizencore.listeners;

import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.block.data.Levelled;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerInteractEvent;
import org.bukkit.inventory.ItemStack;

public class ComposterInteractEvent implements Listener {

    @EventHandler
    public void PlayerInteractComposterEvent(PlayerInteractEvent event){

        Block block = event.getClickedBlock();
        if (block == null) return;
        Material material = block.getType();
        if (!material.equals(Material.COMPOSTER)) return;
        Levelled composter = (Levelled) block.getBlockData();
        Player player = event.getPlayer();

        if (!(composter.getLevel() == (composter.getMaximumLevel() - 1))){

            if (!player.getInventory().getItemInMainHand().getType().toString().contains("SAPLING")) return;

            if (player.getInventory().getItemInMainHand().getAmount() == 1){

                player.getInventory().getItemInMainHand().setAmount(0);

            } else {

                player.getInventory().getItemInMainHand().setAmount(player.getInventory().getItemInMainHand().getAmount() - 1);

            }

            event.setCancelled(true);
            composter.setLevel(composter.getLevel() + 1);
            block.setBlockData(composter);

        } else if (composter.getLevel() == composter.getMaximumLevel()){

            if (player.isSneaking()) return;
            event.setCancelled(true);
            block.getWorld().dropItemNaturally(block.getLocation(), new ItemStack(Material.DIRT, 1));
            composter.setLevel(0);
            block.setBlockData(composter);

        }

    }
}

File: src/main/java/me/waiaf/skytizencore/listeners/ComposterInteractEvent.kt
package me.waiaf.skytizencore.listeners

import org.bukkit.Material
import org.bukkit.block.data.Levelled
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerInteractEvent
import org.bukkit.inventory.ItemStack

class ComposterInteractEvent : Listener {
    @EventHandler
    fun playerInteractComposterEvent(event: PlayerInteractEvent) {
        val block = event.clickedBlock ?: return
        val material = block.type
        if (material != Material.COMPOSTER) return
        val composter = block.blockData as Levelled
        val player = event.player
        if (composter.level != composter.maximumLevel - 1) {
            if (!player.inventory.itemInMainHand.type.toString().contains("SAPLING")) return
            if (player.inventory.itemInMainHand.amount == 1) {
                player.inventory.itemInMainHand.amount = 0
            } else {
                player.inventory.itemInMainHand.amount = player.inventory.itemInMainHand.amount - 1
            }
            event.isCancelled = true
            composter.level = composter.level + 1
            block.blockData = composter
        } else if (composter.level == composter.maximumLevel) {
            if (player.isSneaking) return
            event.isCancelled = true
            block.world.dropItemNaturally(block.location, ItemStack(Material.DIRT, 1))
            composter.level = 0
            block.blockData = composter
        }
    }
}